FreeRTOS/Test/CMock/smp/single_priority_timeslice_thread/single_priority_timeslice_thread_utest.c

```diff
@@ -0,0 +1,170 @@
+{
+    commonTearDown();
+}
+
+/*! called at the beginning of the whole suite */
+void suiteSetUp()
+{
+}
+
+/*! called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* =============================  HELPER FUNCTIONS  ========================= */
+void vApplicationMinimalIdleHook( void )
+{
+    printf( "Minimal idle hook called\r\n" );
+    pthread_exit( NULL );
+}
+
+/* =============================  STATIC FUNCTIONS  ========================= */
+
+static void * task_thread_function( void * args )
+{
+    void * pvParameters = NULL;
+
+    /* Setup */
+    portTASK_FUNCTION( prvMinimalIdleTask, pvParameters );
+    
+    /* API Call */
+    prvMinimalIdleTask( pvParameters );
+
+    return NULL;
+}
+
+/* ==============================  Test Cases  ============================== */
+
+/*
+Coverage for:
+    static portTASK_FUNCTION( prvMinimalIdleTask, pvParameters )
+    
+    requires you to create a thread and kill it, for an idle task is eternal.
+*/
+void test_prvIddleTask_Expected_time( void )
+{
+    TaskHandle_t xTaskHandles[configNUMBER_OF_CORES + 1] = { NULL };
+    uint32_t i, retVal ;
+    pthread_t thread_id;
+
+    /* Create configNUMBER_OF_CORES tasks of equal priority */
+    for (i = 0; i < (configNUMBER_OF_CORES); i++) {
+        xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 2, &xTaskHandles[i] );
+    }
+
+    // /* Create a single equal priority task */   
+    xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 2, &xTaskHandles[0] );
+    
+    vTaskStartScheduler();
+    //Necessary to trigger another function with porttask()
+    vTaskDelete(xTaskHandles[0]);
+
+    /* API Call */
+    pthread_create( &thread_id, NULL, &task_thread_function, NULL );
+    pthread_join( thread_id, ( void ** ) &retVal );
+    
+    for (i = 1; i < (configNUMBER_OF_CORES); i++) {
+        vTaskDelete(xTaskHandles[i]);
+    }
+}
+
+/*
+Coverage for:
+    static portTASK_FUNCTION( prvMinimalIdleTask, pvParameters )
+    
+    for the condition when a task that is sharing the idle priority is 
+    ready to run then the idle task is yielded before the end of the timeslice
+    
+    programmatically:
+        if( listCURRENT_LIST_LENGTH( &( pxReadyTasksLists[ tskIDLE_PRIORITY ] ) ) > ( UBaseType_t ) configNUMBER_OF_CORES ) = True
+
+*/
+void test_prvIddleTask_Expected_time_more_task( void )
+{
+    TaskHandle_t xTaskHandles[configNUMBER_OF_CORES+1] = { NULL };
+    uint32_t retVal ;
+    pthread_t thread_id;
+
+    /* As configNUMBER_OF_CORES idle tasks are already created by FreeRTOS kernel 
+    *  we only need to create one extra to have additonal equal priorty task */
+    xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, tskIDLE_PRIORITY, &xTaskHandles[configNUMBER_OF_CORES] );
+    
+    vTaskStartScheduler();
+
+    /* API Call */
+    pthread_create( &thread_id, NULL, &task_thread_function, NULL );
```

Comment:
This was discussed in an an offline meeting, but I believe we want to make kernel side changes so that we can explicitly iterate and escape from thread functions (or any infinite loops), rather than use pthread to spawn a separate task.